src/Core/Style.js

```diff
@@ -20,30 +20,31 @@ function baseAltitudeDefault(properties, ctx) {
 }
 
 export function readExpression(property, ctx) {
-    if (property != undefined) {
-        if (property.expression) {
-            return property.expression.evaluate(ctx);
-        } else if (property.stops) {
-            for (let i = property.stops.length - 1; i >= 0; i--) {
-                const stop = property.stops[i];
-
-                if (ctx.zoom >= stop[0]) {
-                    return stop[1];
-                }
+    if (property.expression) {
+        return property.expression.evaluate(ctx);
+    }
+    if (property.stops) {
+        const stops = property.stops;
+        property = property.stops[0][1];
+        for (let i = stops.length - 1; i >= 0; i--) {
+            const stop = stops[i];
+
+            if (ctx.zoom >= stop[0]) {
+                property = stop[1];
+                break;
             }
```

Comment:
Can `stops` be empty ?
We could do:
```suggestion
        const stops = property.stops;
       
        for (let i = stops.length - 1; i >= 0; i--) {
            const stop = stops[i];
                property = stop[1];
            if (ctx.zoom >= stop[0]) {

                break;
            }
```
WDYT ?

Reply:
I don't think stop can be empty but we never know.

I do'tn know if waht you propose really doesn't change anything. I fell lik ein cas we have a lot of stops your might be a bit slower as we will re-affect property at each step instea of only at start and at end. But i might be wrong